### PR TITLE
 Fix the redirect issue if user not acting as business

### PR DESCRIPTION
**Pull request by Henry0422**
Fix issue: https://nanopay.atlassian.net/browse/CPF-1116

It might fix: https://nanopay.atlassian.net/browse/CPF-883

### CODE REVIEW

**Comment by Henry0422** [comment]
@mcarcaso @TW80000 Can you review this PR?

**Review by TW80000 — APPROVED** [review]

File: src/foam/nanos/controller/ApplicationController.js [hunk 1]
```diff
@@ -183,18 +183,19 @@ foam.CLASS({
     function init() {
       this.SUPER();
       var self = this;
+
+      window.onpopstate = async function(event) {
+        var hid = location.hash.substr(1);
+        if ( hid ) {
+          var menu = await this.client.menuDAO.find(hid);
```
Comment (TW80000) [inline src/foam/nanos/controller/ApplicationController.js]: We might want to add a null check to make sure `this.client` is set here.

Comment (TW80000) [inline src/foam/nanos/controller/ApplicationController.js]: We might want to add a null check to make sure `this.client` is set here.

**Comment by TW80000** [comment]
Whoops, didn't realize this was FOAM. Let me know if you want me to open a PR reverting this.

**Pull request merged**